backend/src/controllers/OrderController.ts

```diff
@@ -0,0 +1,110 @@
+import { Request, Response } from 'express';
+import Stripe from 'stripe';
+import Restaurant, { MenuItemType } from '../models/restaurant';
+
+const STRIPE = new Stripe(process.env.STRIPE_API_KEY as string);
+const FRONTEND_URL = process.env.FRONTEND_URL as string;
+
+type CheckoutSessionRequest = {
+  cartItems: {
+    menuItemId: string;
+    quantity: number;
+  }[];
+  deliveryDetails: {
+    email: string;
+    name: string;
+    addressLine1: string;
+    city: string;
+  };
+  restaurantId: string;
+};
+
+const createCheckoutSession = async (req: Request, res: Response) => {
+  try {
+    const checkoutSessionRequest: CheckoutSessionRequest = req.body as CheckoutSessionRequest;
+    const restaurant = await Restaurant.findById(checkoutSessionRequest.restaurantId);
+    if (!restaurant) {
+      return res.status(404).send('Restaurant not found');
+    }
+
+    const lineItems = await createLineItems(checkoutSessionRequest, restaurant.menuItems);
+    const session = await createSession(
+      lineItems,
+      'TEST_ORDER_ID',
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/46#issuecomment-2423279657>#e57e5d</a><span> - 10/18/2024, 09:54 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The 'TEST_ORDER_ID' is still being used as a placeholder, which is not suitable for production and may cause issues with order tracking and management.
</div>
<div id="fix">
<b>Fix:</b> Replace 'TEST_ORDER_ID' with a function call that generates a unique order ID. This could be done using a UUID generator or a database sequence.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -30,7 +30,7 @@
      const lineItems = await createLineItems(checkoutSessionRequest, restaurant.menuItems);
      const session = await createSession(
        lineItems,
 -      'TEST_ORDER_ID',
 +      generateUniqueOrderId(),
        restaurant.deliveryPrice,
        restaurant._id.toString(),
      );
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged